lib/kennel/id_map.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+module Kennel
+  class IdMap
+    NEW = :new
+
+    def initialize
+      @map = Hash.new { |h, k| h[k] = {} }
+    end
+
+    def add(type, tracking_id, id)
+      @map[type][tracking_id] = id
```

Comment:
FYI this would also work, but would create more bogus array objects
```suggestion
      @map[[type, tracking_id]] = id
```